src/bun.js/test/expect.zig

```diff
@@ -220,6 +220,16 @@ pub const Expect = struct {
 
     pub fn getSnapshotName(this: *Expect, allocator: std.mem.Allocator, hint: string) ![]const u8 {
         const test_name = this.scope.tests.items[this.test_id].label;
+        var name_builder = MutableString.initEmpty(allocator);
+
+        for (test_name) |value| {
```

Comment:
1) Since this is an edgecase, Let's avoid the allocation unless we know for sure there's a backtick quote character:
```
var escaped_name = test_name;
  if (strings.indexOfChar(test_name, '`')) |start_i| {
      ...
  }
}
```

2) Instead of copying the string byte-by-byte, which is slow, let's find the index of the next backtick character, copy everything before that index, and then copy the escape + backtick character. And then don't forget to append at the end.

Reply:
For point 1, why would we use shadowing for the if block?